### ISSUE TITLE
How to define a saddle objective function ?

### ISSUE THREAD

**Issue by ktiwari9** [CalebEverett/pysgd#1]
Hi there,
I am trying to implement the adam for a saddle function like the one drawn using : [https://commons.wikimedia.org/wiki/File:Saddle_point.svg](url)   . I am having trouble translating this 3-d function as an objective for your framework. Could you please help out with this ?

**Comment by CalebEverett** [comment]
Sure, I couldn't get to the link you posted, though.

**Comment by ktiwari9** [comment]
I think you can copy-paste the link manually then it works. Directly clicking doesnt open it for some reason.

**Comment by CalebEverett** [comment]
It looks like the Saddle Function is z = x2 - y2 so you could add a new
gradient and cost function module that mirrors the form of
https://github.com/CalebEverett/pysgd/blob/master/pysgd/objectives/stab_tang.py
.

But, without knowing your use case, that function doesn't really have any
minima, so it won't converge.

**Comment by ktiwari9** [comment]
Yeah I was trying to add this saddle function to the objective functions folder but I am having trouble understanding what is the format of the grad_fun and cost_fun. I mean the elements of grad_fun for saddle function should be {derivative with respect to x, derivative with respect to y,derivative with respect to z } but in your case you have a ZERO and a theta in the list which confuses me. Also in the cost function, everything is in terms of theta so in saddle function case, theta = [x,y,z] and the objective is still defined in terms of theta ?  

What I want to try is to see if Adam or Adagrad get stuck in a loop in this saddle function since there is no minima. Theretically, they should not diverge but this is what I wanted to see that is why I want to add this new objective function.
